I see a build failure, but it looks to be related to stepper and not this PR?
```
  client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/index.tsx(86,34): error TS2339: Property 'hasActiveSiteFeature' does not exist on type 'SelectFromMap<typeof import("/home/teamcity-2/buildAgent/work/c4a9d5b38c1dacde/packages/data-stores/dist/types/site/selectors")>'.
```

--- a/client/my-sites/scan/wpcom-upsell.tsx
+++ b/client/my-sites/scan/wpcom-upsell.tsx
@@ -1,7 +1,10 @@
-import { isPersonalPlan, isPremiumPlan } from '@automattic/calypso-products';
+import {
+	WPCOM_FEATURES_BACKUPS,
+	WPCOM_FEATURES_FULL_ACTIVITY_LOG,
+} from '@automattic/calypso-products';
 import { Gridicon } from '@automattic/components';
 import { useTranslate } from 'i18n-calypso';
-import { useMemo, ReactElement } from 'react';
+import { ReactElement } from 'react';
 import { useSelector } from 'react-redux';
 import JetpackScanSVG from 'calypso/assets/images/illustrations/jetpack-scan.svg';
 import DocumentHead from 'calypso/components/data/document-head';
@@ -13,13 +16,23 @@ import PromoCard from 'calypso/components/promo-section/promo-card';
 import PromoCardCTA from 'calypso/components/promo-section/promo-card/cta';
 import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
 import useTrackCallback from 'calypso/lib/jetpack/use-track-callback';
-import { getSitePlan } from 'calypso/state/sites/selectors';
+import siteHasFeature from 'calypso/state/selectors/site-has-feature';
 import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
 
 import './style.scss';
 
 export default function WPCOMScanUpsellPage(): ReactElement {
 	const translate = useTranslate();
+	const onUpgradeClick = useTrackCallback( undefined, 'calypso_jetpack_scan_business_upsell' );
+	const siteSlug = useSelector( getSelectedSiteSlug );
+	const siteId = useSelector( getSelectedSiteId );
+	const hasBackups = useSelector( ( state ) =>
+		siteHasFeature( state, siteId, WPCOM_FEATURES_BACKUPS )
+	);
+	const hasFullActivityLog = useSelector( ( state ) =>
+		siteHasFeature( state, siteId, WPCOM_FEATURES_FULL_ACTIVITY_LOG )
+	);
+
 	const promos = [
 		{
 			title: translate( 'Jetpack Backup' ),
@@ -28,6 +41,7 @@ export default function WPCOMScanUpsellPage(): ReactElement {
 					'to restore it to any previous state, and export it at any time.'
 			),
 			image: <Gridicon icon="cloud-upload" className="scan__upsell-icon" />,
+			isShown: ! hasBackups,
 		},
 		{
 			title: translate( 'Activity Log' ),
@@ -35,22 +49,12 @@ export default function WPCOMScanUpsellPage(): ReactElement {
 				'A complete record of everything that happens on your site, with history that spans over 30 days.'
 			),
 			image: <Gridicon icon="history" className="scan__upsell-icon" />,
+			isShown: ! hasFullActivityLog,
 		},
 	];
 
-	const onUpgradeClick = useTrackCallback( undefined, 'calypso_jetpack_scan_business_upsell' );
-	const siteSlug = useSelector( getSelectedSiteSlug );
-	const siteId = useSelector( getSelectedSiteId );
-	const { product_slug: planSlug = '' } =
-		useSelector( ( state ) => getSitePlan( state, siteId ) ) ?? {};
-
-	// Don't show the Activity Log promo for Personal or Premium plan owners.
-	const filteredPromos: PromoSectionProps = useMemo( () => {
-		if ( isPersonalPlan( planSlug ) || isPremiumPlan( planSlug ) ) {
-			return { promos: [ promos[ 0 ] ] };
-		}
-		return { promos };
-	}, [ planSlug ] );
+	// Only show promos for features the blog does not already have.
+	const filteredPromos: PromoSectionProps = { promos: promos.filter( ( p ) => p.isShown ) };
 
 	return (
 		<Main className="scan scan__wpcom-upsell">
@@ -87,9 +91,12 @@ export default function WPCOMScanUpsellPage(): ReactElement {
 				/>
 			</PromoCard>
 
-			<h2 className="scan__subheader">{ translate( 'Also included in the Pro Plan' ) }</h2>
-
-			<PromoSection { ...filteredPromos } />
+			{ filteredPromos.promos.length > 0 && (
+				<>
+					<h2 className="scan__subheader">{ translate( 'Also included in the Pro Plan' ) }</h2>
+					<PromoSection { ...filteredPromos } />
+				</>
+			) }
 
 			<WhatIsJetpack />
 		</Main>

